opm/simulators/flow/Banners.cpp

```diff
@@ -0,0 +1,124 @@
+/*
+  Copyright 2013, 2014, 2015 SINTEF ICT, Applied Mathematics.
+  Copyright 2014 Dr. Blatt - HPC-Simulation-Software & Services
+  Copyright 2015 IRIS AS
+  Copyright 2014 STATOIL ASA.
+
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <config.h>
+#include <opm/simulators/flow/Banners.hpp>
+
+#include <opm/common/OpmLog/OpmLog.hpp>
+
+#include <opm/simulators/timestepping/SimulatorReport.hpp>
+#include <opm/simulators/utils/moduleVersion.hpp>
+
+#include <fmt/format.h>
+
+#include <iomanip>
+#include <iostream>
+#include <sstream>
+#include <sys/utsname.h>
+#include <thread>
+#include <unistd.h>
+
+namespace {
+
+static unsigned long long getTotalSystemMemory()
+{
+    long pages = sysconf(_SC_PHYS_PAGES);
+    long page_size = sysconf(_SC_PAGE_SIZE);
+    return pages * page_size;
+}
+
+}
+
+namespace Opm {
+
+void printPRTHeader(const std::string& parameters)
+{
+    const std::string version = moduleVersion();
+    const double megabyte = 1024 * 1024;
+    unsigned num_cpu = std::thread::hardware_concurrency();
+    struct utsname arch;
+    const char* user = getlogin();
+    time_t now = std::time(0);
+    struct tm  tstruct;
+    char      tmstr[80];
+    tstruct = *localtime(&now);
+    strftime(tmstr, sizeof(tmstr), "%d-%m-%Y at %X", &tstruct);
```

Comment:
We should include `<time.h>` for `time_t`, `struct tm`, `localtime()`, and `strftime()`.  Optionally we could include `<ctime>` and reference the `std::` symbols instead.

Reply:
done (using std:: )